web-api/src/personalization/routes.rs

```diff
@@ -464,27 +382,119 @@ pub(crate) async fn personalize_documents_by(
 }
 
 #[derive(Deserialize)]
-struct SemanticSearchQuery {
+struct UnvalidatedInputUser {
+    #[serde(default)]
+    id: Option<String>,
+    #[serde(default)]
+    history: Option<Vec<UnvalidatedHistoryEntry>>,
+}
```

Comment:
:art: since all fields are deserialized as defaults, we can just annotate it for the struct instead
```suggestion
#[derive(Default, Deserialize)]
#[serde(default)]
struct UnvalidatedInputUser {
    id: Option<String>,
    history: Option<Vec<UnvalidatedHistoryEntry>>,
}
```